pkg/gateway/server/oauth_apps.go

```diff
@@ -746,5 +746,21 @@ func convertOAuthAppRegistrationToOAuthApp(app v1.OAuthApp, baseURL string) type
 
 func getOAuthAppFromName(apiContext api.Context) (*v1.OAuthApp, error) {
 	var oauthApp v1.OAuthApp
-	return &oauthApp, apiContext.Get(&oauthApp, apiContext.PathValue("id"))
+	id := apiContext.PathValue("id")
+	if !strings.HasPrefix(id, system.OAuthAppPrefix) {
+		var oauthApps v1.OAuthAppList
+		if err := apiContext.List(&oauthApps, &kclient.ListOptions{
+			FieldSelector: fields.SelectorFromSet(selectors.RemoveEmpty(map[string]string{
+				"spec.manifest.alias": id,
+			})),
+			Namespace: apiContext.Namespace(),
+		}); err != nil {
+			return nil, err
+		}
+		if len(oauthApps.Items) == 0 {
+			return nil, fmt.Errorf("oauth app %s not found", id)
+		}
+		return &oauthApps.Items[0], nil
+	}
+	return &oauthApp, apiContext.Get(&oauthApp, id)
```

Comment:
Can we use this: https://github.com/thedadams/obot/blob/otel/pkg/alias/get.go#L19?

Reply:
I think we can remove the alias later. This is for backwards compabities so we don't have to update existing oauth apps to use id instead of alias